lib/Sema/TypeCheckAttr.cpp

```diff
@@ -7279,6 +7279,13 @@ void AttributeChecker::visitUnsafeInheritExecutorAttr(
   auto fn = cast<FuncDecl>(D);
   if (!fn->isAsyncContext()) {
     diagnose(attr->getLocation(), diag::inherits_executor_without_async);
+  } else {
+    bool inConcurrencyModule = D->getDeclContext()->getParentModule()->getName()
+        .str().equals("_Concurrency");
+    auto diag = fn->diagnose(diag::unsafe_inherits_executor_deprecated);
+    diag.warnUntilSwiftVersion(6);
```

Comment:
I'm a bit worried about making it a hard error since I've not been so confident in all the situations with isolation inference and #isolation actually working just yet... but yes, long term this should be an error. 

Perhaps we just have to take the risk.

Reply:
Right. The combination of unsafe / underscored / does not compose with its replacement is what pushed me to outright ban it in Swift 6, but we could back off from this.

Reply:
It’s probably fine, worst that can happen in oss is delaying 6 mode adoption in case #isolation still has some bugs. That’s okey I think, worth pushing folks off the underscored thing because it’s so broken